pkg/authentication/x509/x509_test.go

```diff
@@ -0,0 +1,274 @@
+
+	err = ensureAgent(&auth, secrets, mdb)
+	assert.Error(t, err)
+
+	auth = automationconfig.Auth{}
+	agentSecret = CreateAgentCertificateSecret("tls.crt", mdb, true)
+	secrets = mocks.NewMockedSecretGetUpdateCreateDeleter(agentSecret)
+
+	err = ensureAgent(&auth, secrets, mdb)
+	assert.Error(t, err)
+
+	auth = automationconfig.Auth{}
+	agentSecret = CreateAgentCertificateSecret("tls.crt", mdb, false)
+	secrets = mocks.NewMockedSecretGetUpdateCreateDeleter(agentSecret)
+
+	err = ensureAgent(&auth, secrets, mdb)
+	assert.NoError(t, err)
+}
+
+func Test_convertMongoDBResourceUsersToAutomationConfigUsers(t *testing.T) {
+	type args struct {
+		mdb authtypes.Configurable
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []automationconfig.MongoDBUser
+		wantErr bool
+	}{
+		{
+			name: "Only x.509 users",
+			args: args{mdb: buildX509Configurable("mongodb", mocks.BuildX509MongoDBUser("my-user-1"), mocks.BuildX509MongoDBUser("my-user-2"))},
+			want: []automationconfig.MongoDBUser{
+				{
+					Mechanisms: []string{},
+					Roles: []automationconfig.Role{
+						{
+							Role:     "readWrite",
+							Database: "admin",
+						},
+						{
+							Role:     "clusterAdmin",
+							Database: "admin",
+						},
+					},
+					Username:                   "CN=my-user-1,OU=organizationalunit,O=organization",
+					Database:                   "$external",
+					AuthenticationRestrictions: []string{},
+				},
+				{
+					Mechanisms: []string{},
+					Roles: []automationconfig.Role{
+						{
+							Role:     "readWrite",
+							Database: "admin",
+						},
+						{
+							Role:     "clusterAdmin",
+							Database: "admin",
+						},
+					},
+					Username:                   "CN=my-user-2,OU=organizationalunit,O=organization",
+					Database:                   "$external",
+					AuthenticationRestrictions: []string{},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Only SCRAM users",
+			args:    args{mdb: buildX509Configurable("mongodb", mocks.BuildScramMongoDBUser("my-user-1"), mocks.BuildScramMongoDBUser("my-user-2"))},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "X.509 and SCRAM users",
+			args: args{mdb: buildX509Configurable("mongodb", mocks.BuildX509MongoDBUser("my-user-1"), mocks.BuildScramMongoDBUser("my-user-2"))},
+			want: []automationconfig.MongoDBUser{
+				{
+					Mechanisms: []string{},
+					Roles: []automationconfig.Role{
+						{
+							Role:     "readWrite",
+							Database: "admin",
+						},
+						{
+							Role:     "clusterAdmin",
+							Database: "admin",
+						},
+					},
+					Username:                   "CN=my-user-1,OU=organizationalunit,O=organization",
+					Database:                   "$external",
+					AuthenticationRestrictions: []string{},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := convertMongoDBResourceUsersToAutomationConfigUsers(tt.args.mdb)
+			if (err != nil) != tt.wantErr {
```

Comment:
this does not look idiomatic, can't it be just a direct comparison between err and wanterr?

Reply:
I noticed that the method tested there always returns a nil error so I will remove tests for that.